File: device_information.py
import platform
import psutil
from time import sleep


print("\nCarregando as informações...")

informacoes = {}

# - Nome do sistema
nome_sistema = platform.system()
informacoes["Nome do Sistema"] = nome_sistema

# - Versão do sistema
versao_sistema = platform.platform()
platform.release()
informacoes["Versão do Sistema"] = versao_sistema

# - Nome do processador
processador = platform.processor()
informacoes["Nome do Processador"] = processador

# - Detalhes da arquitetura
arquitetura = platform.architecture()
informacoes["Arquitetura"] = arquitetura

# - Máquina
maquina = platform.machine()
informacoes["Máquina"] = maquina

# - Rede
rede = platform.node()
informacoes["Rede"] = rede

# - Uso da CPU (intervalo de 4 segundos)
porcentagem_cpu = psutil.cpu_percent(4)
informacoes["Porcentagem da CPU"] = porcentagem_cpu

# - Uso de memória RAM
porcentagem_ram = psutil.virtual_memory()[2]
informacoes["Porcentagem de RAM"] = porcentagem_ram

# - Bateria
bateria = psutil.sensors_battery()

# Porcentagem:
porcentagem_bateria = str(bateria.percent)
# Está carregando ou não
carregando = bateria.power_plugged

plugado = "ligado à bateria"
if carregando:
    plugado = "ligado à tomada"
else:
    plugado = "ligado à bateria"

informacoes["Porcentagem de Bateria"] = porcentagem_bateria
informacoes["Modo de Energia"] = plugado

print("\nInformações:")
for item, valor in informacoes.items():
    print(f"- {item}: {valor}")
    sleep(1)

print("\nProcesso finalizado.")
print(informacoes)

File: device_information.js
const os = require("os")


console.log("\nCarregando as informações...")

let informacoes = {}

// - Arquitetura
informacoes["Arquitetura"] = os.arch()

// - Plataforma
informacoes["Plataforma"] = os.platform()

// - Detalhes da CPU
informacoes["Detalhes da CPU"] = os.cpus()

// - Quantidade da CPU
informacoes["Quantidade de CPU"] = os.cpus().length

// - Memória total (Gb)
const memoria_total = (os.totalmem()/1024)/1024
informacoes["Memória Total"] = memoria_total

// - Memória livre (Gb)
const memoria_livre = (os.freemem()/1024)/1024
informacoes["Memória Livre"] = memoria_livre

// Porcentagem de memória livre (%)
const porcentagem = (memoria_livre/memoria_total)*100
informacoes["Porcentagem de Memória Livre"] = porcentagem

// Tempo da máquina ligada (horas)
const tempo = (os.uptime()/60)/60
informacoes["Uptime"] = tempo

// Interfaces de redes
informacoes["Interfaces de Redes"] = os.networkInterfaces()

// Caminho do usuário
informacoes["Caminho do Usuário"] = os.homedir()

// Versão do Sistema Operacional
informacoes["Versão do Sistema Operacional"] = os.release()

// Sistema Operacional
informacoes["Sistema Operacional"] = os.type()

// Nome do Host
// informacoes["Hostname"] = os.hostname()

console.log("\nInformações:")
for (const item in informacoes){
    console.log(`- ${item}: ${informacoes[item]}`)
}